File: Unse/Features/Onboarding/OnboardingView.swift
import SwiftUI

// MARK: - OnboardingView

struct OnboardingView: View {
    @State private var slide = 0
    @State private var showBirthInfo = false

    private let slides: [OnboardSlide] = [
        .init(eyebrow: "매일 아침",
              titleA: "당신의", titleB: "행운 색을",
              subtitle: "사주 8글자에서 풀어낸\n오늘의 색·방향·시간",
              art: .colors),
        .init(eyebrow: "AI 해설",
              titleA: "진짜 사주를", titleB: "풀어드려요",
              subtitle: "카톡 운세 말고,\n진짜 명리학 기반 풀이",
              art: .chat),
        .init(eyebrow: "평생의 흐름",
              titleA: "지금이", titleB: "어떤 시기인지",
              subtitle: "대운 그래프로 보는\n인생의 정점과 저점",
              art: .timeline),
    ]

    var body: some View {
        NavigationStack {
            VStack(spacing: 0) {
                skipButton
                artPager
                copySection
            }
            .background(Color.bg.ignoresSafeArea())
            .navigationDestination(isPresented: $showBirthInfo) {
                BirthInfoView()
            }
        }
    }

    private var skipButton: some View {
        HStack {
            Spacer()
            Button("건너뛰기") { showBirthInfo = true }
                .font(.pretendard(15, .medium))
                .foregroundStyle(.ink3)
                .padding(.horizontal, Spacing.lg)
                .padding(.top, Spacing.sm)
        }
    }

    private var artPager: some View {
        TabView(selection: $slide) {
            ForEach(slides.indices, id: \.self) { i in
                OnboardArt(kind: slides[i].art)
                    .tag(i)
                    .padding(.horizontal, Spacing.xxxl)
            }
        }
        .tabViewStyle(.page(indexDisplayMode: .never))
        .frame(maxHeight: .infinity)
    }

    private var copySection: some View {
        VStack(alignment: .leading, spacing: 0) {
            Tag(text: slides[slide].eyebrow)
                .padding(.bottom, 14)

            Text("\(slides[slide].titleA)\n\(slides[slide].titleB)")
                .font(.serifKR(34, .semibold))
                .foregroundStyle(.ink1)
                .lineSpacing(8)
                .fixedSize(horizontal: false, vertical: true)

            Text(slides[slide].subtitle)
                .font(.pretendard(15))
                .foregroundStyle(.ink2)
                .lineSpacing(6)
                .padding(.top, 14)
                .fixedSize(horizontal: false, vertical: true)

            PageDots(count: slides.count, current: slide)
                .frame(maxWidth: .infinity)
                .padding(.top, 28)
                .padding(.bottom, 20)

            PrimaryButton(title: slide < 2 ? "다음" : "시작하기") {
                if slide < 2 {
                    withAnimation { slide += 1 }
                } else {
                    showBirthInfo = true
                }
            }
        }
        .padding(.horizontal, Spacing.xxxl)
        .padding(.bottom, 32)
    }
}

// MARK: - OnboardSlide

struct OnboardSlide {
    let eyebrow: String
    let titleA: String
    let titleB: String
    let subtitle: String
    let art: ArtKind

    enum ArtKind { case colors, chat, timeline }
}

// MARK: - OnboardArt

struct OnboardArt: View {
    let kind: OnboardSlide.ArtKind

    var body: some View {
        switch kind {
        case .colors:   OnboardArtColors()
        case .chat:     OnboardArtChat()
        case .timeline: OnboardArtTimeline()
        }
    }
}

// MARK: - OnboardArtColors (floating color cards)

struct OnboardArtColors: View {
    private let cards: [(color: Color, label: String, angle: Double, x: CGFloat, y: CGFloat)] = [
        (.lavender, "라벤더", -12, -65,  -60),
        (.peach,    "피치",     6,  40,  -15),
        (.mint,     "민트",    -8, -35,   60),
        (.cream,    "크림",    10,  55,   65),
    ]

    var body: some View {
        ZStack {
            ForEach(cards.indices, id: \.self) { i in
                let c = cards[i]
                RoundedRectangle(cornerRadius: 22)
                    .fill(c.color)
                    .frame(width: 130, height: 170)
                    .overlay(alignment: .bottomLeading) {
                        VStack(alignment: .leading, spacing: 2) {
                            Text("TODAY")
                                .font(.pretendard(11, .semibold))
                                .foregroundStyle(.black.opacity(0.5))
                                .tracking(1)
                            Text(c.label)
                                .font(.serifKR(20, .semibold))
                                .foregroundStyle(.ink1)
                        }
                        .padding(14)
                    }
                    .shadow(color: .black.opacity(0.08), radius: 30, y: 12)
                    .rotationEffect(.degrees(c.angle))
                    .offset(x: c.x, y: c.y)
            }
        }
        .frame(width: 280, height: 280)
    }
}

// MARK: - OnboardArtChat

struct OnboardArtChat: View {
    var body: some View {
        VStack(spacing: 10) {
            // User bubble
            HStack {
                Spacer()
                Text("올해 이직해도 괜찮을까요?")
                    .font(.pretendard(14))
                    .foregroundStyle(.white)
                    .padding(.horizontal, 16).padding(.vertical, 12)
                    .background(Color.ink1)
                    .clipShape(RoundedCorner(radius: 20, corners: [.topLeft, .topRight, .bottomLeft]))
            }

            // Assistant bubble
            HStack {
                Text("**戊土 일간**에\n**木 대운**이 들어오는 시기네요.\n새 시작이 잘 풀리는 흐름으로 해석됩니다.")
                    .font(.pretendard(14))
                    .foregroundStyle(.ink1)
                    .lineSpacing(4)
                    .padding(.horizontal, 18).padding(.vertical, 14)
                    .background(Color.surface)
                    .clipShape(RoundedCorner(radius: 20, corners: [.topLeft, .topRight, .bottomRight]))
                    .shadow(color: .black.opacity(0.04), radius: 16, y: 4)
                Spacer()
            }

            // Typing indicator
            HStack {
                HStack(spacing: 4) {
                    ForEach(0..<3, id: \.self) { i in
                        Circle().fill(Color.lavenderDeep).frame(width: 6, height: 6)
                            .opacity(Double(3 - i) / 3.0)
                    }
                }
                .padding(.horizontal, 16).padding(.vertical, 12)
                .background(Color.surface)
                .clipShape(RoundedRectangle(cornerRadius: 20))
                .shadow(color: .black.opacity(0.04), radius: 16, y: 4)
                Spacer()
            }
        }
        .frame(maxWidth: 280)
    }
}

// MARK: - OnboardArtTimeline

struct OnboardArtTimeline: View {
    var body: some View {
        Canvas { ctx, size in
            let w = size.width, h = size.height

            // Zone backgrounds
            let zones: [(Color, CGFloat)] = [(.lavenderSoft, 0.3), (.peachSoft, 0.4), (.mintSoft, 0.3)]
            for (i, (color, opacity)) in zones.enumerated() {
                let x = CGFloat(i) * w / 3
                let rect = CGRect(x: x, y: 20, width: w/3, height: h-40)
                ctx.fill(Path(rect), with: .color(color.opacity(opacity)))
            }

            // Lines (simplified sine-like paths)
            func drawLine(_ points: [(CGFloat, CGFloat)], color: Color) {
                var path = Path()
                path.move(to: CGPoint(x: points[0].0 * w, y: points[0].1 * h))
                for p in points.dropFirst() {
                    path.addLine(to: CGPoint(x: p.0 * w, y: p.1 * h))
                }
                ctx.stroke(path, with: .color(color), lineWidth: 2.5)
            }
            drawLine([(0,0.6),(0.3,0.4),(0.6,0.3),(1.0,0.5)], color: .lavenderDeep)
            drawLine([(0,0.7),(0.3,0.65),(0.6,0.45),(1.0,0.65)], color: .peachDeep)
            drawLine([(0,0.5),(0.3,0.55),(0.6,0.35),(1.0,0.45)], color: .mintDeep)

            // Current position marker
            let cx = w * 0.43
            ctx.stroke(Path { p in p.move(to: CGPoint(x: cx, y: 20)); p.addLine(to: CGPoint(x: cx, y: h-20)) },
                       with: .color(.ink1), style: StrokeStyle(lineWidth: 1.5, dash: [3]))
            ctx.fill(Path(ellipseIn: CGRect(x: cx-5, y: h*0.3-5, width: 10, height: 10)), with: .color(.ink1))
        }
        .frame(maxWidth: 300, maxHeight: 220)
    }
}

// MARK: - Helpers

struct RoundedCorner: Shape {
    var radius: CGFloat
    var corners: UIRectCorner

    func path(in rect: CGRect) -> Path {
        Path(UIBezierPath(
            roundedRect: rect,
            byRoundingCorners: corners,
            cornerRadii: CGSize(width: radius, height: radius)
        ).cgPath)
    }
}

File: android/app/src/main/kotlin/run/mound/unse/ui/onboarding/OnboardingView.kt
package run.mound.unse.ui.onboarding

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.launch
import run.mound.unse.ui.components.PrimaryButton
import run.mound.unse.ui.components.Tag
import run.mound.unse.ui.theme.Bg
import run.mound.unse.ui.theme.Cream
import run.mound.unse.ui.theme.Ink1
import run.mound.unse.ui.theme.Ink2
import run.mound.unse.ui.theme.Ink3
import run.mound.unse.ui.theme.Lavender
import run.mound.unse.ui.theme.LavenderDeep
import run.mound.unse.ui.theme.LavenderSoft
import run.mound.unse.ui.theme.Mint
import run.mound.unse.ui.theme.Peach
import run.mound.unse.ui.theme.Spacing
import run.mound.unse.ui.theme.Surface

/**
 * 하루결 OnboardingView — iOS OnboardingView.swift 1:1.
 * 3 슬라이드: 행운의 색 / AI 해설 / 평생 흐름.
 *
 * 레이아웃: skipButton 위 + artPager (weight 1) + copySection 아래 고정.
 */

private enum class ArtKind { COLORS, CHAT, TIMELINE }

private data class OnboardSlide(
    val eyebrow: String,
    val titleA: String,
    val titleB: String,
    val subtitle: String,
    val art: ArtKind
)

private val slides = listOf(
    OnboardSlide(
        eyebrow = "매일 아침",
        titleA = "오늘의",
        titleB = "행운 색",
        subtitle = "사주 8글자에서 풀어낸\n오늘의 색·방향·시간",
        art = ArtKind.COLORS
    ),
    OnboardSlide(
        eyebrow = "AI 해설",
        titleA = "진짜 사주를",
        titleB = "풀어드려요",
        subtitle = "카톡 운세 말고,\n진짜 명리학 기반 풀이",
        art = ArtKind.CHAT
    ),
    OnboardSlide(
        eyebrow = "평생의 흐름",
        titleA = "지금이",
        titleB = "어떤 시기인지",
        subtitle = "대운 그래프로 보는\n인생의 정점과 저점",
        art = ArtKind.TIMELINE
    )
)

@Composable
fun OnboardingView(onFinish: () -> Unit) {
    val pagerState = rememberPagerState(pageCount = { slides.size })
    val scope = rememberCoroutineScope()

    Column(modifier = Modifier.fillMaxSize().background(Bg)) {
        // 상단 Skip
        Row(
            modifier = Modifier.fillMaxWidth().padding(horizontal = Spacing.lg, vertical = Spacing.sm),
            horizontalArrangement = Arrangement.End
        ) {
            Text(
                text = "건너뛰기",
                color = Ink3,
                style = MaterialTheme.typography.titleMedium,
                modifier = Modifier.clickable { onFinish() }.padding(Spacing.sm)
            )
        }

        // Art Pager — 위쪽 영역 (flex)
        HorizontalPager(
            state = pagerState,
            modifier = Modifier.weight(1f),
            contentPadding = androidx.compose.foundation.layout.PaddingValues(horizontal = Spacing.xxxl)
        ) { page ->
            Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                when (slides[page].art) {
                    ArtKind.COLORS -> ColorCardsArt()
                    ArtKind.CHAT -> ChatArt()
                    ArtKind.TIMELINE -> TimelineArt()
                }
            }
        }

        // Copy 섹션 — 하단 고정
        val s = slides[pagerState.currentPage]
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = Spacing.xxxl, vertical = 0.dp)
                .padding(bottom = 32.dp)
        ) {
            Tag(text = s.eyebrow)
            Spacer(Modifier.height(14.dp))
            Text(
                text = "${s.titleA}\n${s.titleB}",
                style = MaterialTheme.typography.displayLarge,
                color = Ink1,
                lineHeight = 42.sp
            )
            Spacer(Modifier.height(14.dp))
            Text(
                text = s.subtitle,
                color = Ink2,
                style = MaterialTheme.typography.bodyMedium,
                lineHeight = 22.sp
            )

            // Dots
            Row(
                modifier = Modifier.fillMaxWidth().padding(top = 28.dp, bottom = 20.dp),
                horizontalArrangement = Arrangement.Center
            ) {
                repeat(slides.size) { i ->
                    val selected = pagerState.currentPage == i
                    Box(
                        modifier = Modifier
                            .padding(horizontal = 4.dp)
                            .size(if (selected) 24.dp else 8.dp, 8.dp)
                            .clip(RoundedCornerShape(50))
                            .background(if (selected) LavenderDeep else Lavender.copy(alpha = 0.4f))
                    )
                }
            }

            val isLast = pagerState.currentPage == slides.size - 1
            PrimaryButton(title = if (isLast) "시작하기" else "다음") {
                if (isLast) onFinish()
                else scope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) }
            }
        }
    }
}

// MARK: - ColorCardsArt (iOS OnboardArtColors — floating cards with rotation)

@Composable
private fun ColorCardsArt() {
    Box(modifier = Modifier.size(280.dp), contentAlignment = Alignment.Center) {
        FloatingCard("라벤더", Lavender, angle = -12f, x = (-65).dp, y = (-60).dp)
        FloatingCard("피치", Peach, angle = 6f, x = 40.dp, y = (-15).dp)
        FloatingCard("민트", Mint, angle = -8f, x = (-35).dp, y = 60.dp)
        FloatingCard("크림", Cream, angle = 10f, x = 55.dp, y = 65.dp)
    }
}

@Composable
private fun FloatingCard(
    label: String,
    color: Color,
    angle: Float,
    x: androidx.compose.ui.unit.Dp,
    y: androidx.compose.ui.unit.Dp
) {
    Box(
        modifier = Modifier
            .offset(x = x, y = y)
            .rotate(angle)
            .size(130.dp, 170.dp)
            .clip(RoundedCornerShape(22.dp))
            .background(color),
        contentAlignment = Alignment.BottomStart
    ) {
        Column(modifier = Modifier.padding(14.dp)) {
            Text(
                text = "TODAY",
                color = Color.Black.copy(alpha = 0.5f),
                style = MaterialTheme.typography.labelSmall,
                letterSpacing = 1.sp
            )
            Spacer(Modifier.height(2.dp))
            Text(text = label, style = MaterialTheme.typography.titleLarge, color = Ink1)
        }
    }
}

// MARK: - ChatArt (iOS OnboardArtChat — 2 bubbles + typing indicator)

@Composable
private fun ChatArt() {
    Column(
        modifier = Modifier.fillMaxWidth().padding(horizontal = Spacing.lg),
        verticalArrangement = Arrangement.spacedBy(10.dp)
    ) {
        // 사용자 메시지
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.End) {
            Text(
                text = "올해 이직해도 괜찮을까요?",
                color = Color.White,
                style = MaterialTheme.typography.bodyMedium,
                modifier = Modifier
                    .clip(RoundedCornerShape(topStart = 20.dp, topEnd = 20.dp, bottomStart = 20.dp, bottomEnd = 4.dp))
                    .background(Ink1)
                    .padding(horizontal = 16.dp, vertical = 12.dp)
            )
        }
        // AI 응답
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Start) {
            Text(
                text = "戊土 일간에\n木 대운이 들어오는 시기네요.\n새 시작이 잘 풀리는 흐름으로 해석됩니다.",
                color = Ink1,
                style = MaterialTheme.typography.bodyMedium,
                lineHeight = 20.sp,
                modifier = Modifier
                    .clip(RoundedCornerShape(topStart = 20.dp, topEnd = 20.dp, bottomStart = 4.dp, bottomEnd = 20.dp))
                    .background(Surface)
                    .padding(horizontal = 18.dp, vertical = 14.dp)
            )
        }
        // Typing indicator
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Start) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(4.dp),
                modifier = Modifier
                    .clip(RoundedCornerShape(20.dp))
                    .background(Surface)
                    .padding(horizontal = 16.dp, vertical = 12.dp)
            ) {
                repeat(3) { i ->
                    Box(
                        modifier = Modifier
                            .size(6.dp)
                            .clip(RoundedCornerShape(50))
                            .background(LavenderDeep.copy(alpha = (3 - i) / 3f))
                    )
                }
            }
        }
    }
}

// MARK: - TimelineArt (iOS OnboardArtTimeline — zone bars + lines)

@Composable
private fun TimelineArt() {
    Row(
        modifier = Modifier.fillMaxWidth().size(width = 280.dp, height = 200.dp),
        horizontalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterHorizontally),
        verticalAlignment = Alignment.Bottom
    ) {
        // 평생 흐름 막대 — 3개 zone (저 / 평 / 고)
        listOf(60, 80, 110, 140, 120, 100, 80, 60).forEachIndexed { i, h ->
            val color = when {
                i < 3 -> LavenderSoft
                i < 6 -> Peach.copy(alpha = 0.6f)
                else -> Mint.copy(alpha = 0.6f)
            }
            Box(
                modifier = Modifier
                    .size(20.dp, h.dp)
                    .clip(RoundedCornerShape(6.dp))
                    .background(color)
            )
        }
    }
}
